app/models/user.rb

```diff
@@ -418,6 +421,54 @@ def setup(opts)
     self
   end
 
+  def is_pam_account?
+    return false unless Devise.pam_authentication
+    name = pam_get_name
+    return false unless name
+    Rpam2.account(Devise.pam_default_service, name).present?
+  end
+
+  def no_pam_conflict
+    errors.add(:username, 'username conflicts with pam account') if pam_conflict?
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.